src/BloomExe/FontProcessing/FontsApi.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using Bloom.Api;
+using Newtonsoft.Json;
+using System.Collections.Concurrent;
+
+namespace Bloom.FontProcessing
+{
+	public class FontsApi
+	{
+		public const string kApiUrlPart = "fonts/";
+		private static ConcurrentDictionary<string, FontMetadata> _fontNameToMetadata = new ConcurrentDictionary<string, FontMetadata>();
+		private static FontFileFinder _finder;
+		private static object _fontMetadataLock = new object();
+
+		public FontsApi()
+		{
+		}
+
+		public void RegisterWithApiHandler(BloomApiHandler apiHandler)
+		{
+			apiHandler.RegisterEndpointHandler(kApiUrlPart + "names", HandleNamesRequest, false);
+			apiHandler.RegisterEndpointHandler(kApiUrlPart + "metadata", HandleMetadataRequest, false);
+		}
+
+		private void HandleNamesRequest(ApiRequest request)
+		{
+			var list = SortedListOfFontNames();
+			request.ReplyWithJson(JsonConvert.SerializeObject(new { fonts = list }));
+		}
+
+		private void HandleMetadataRequest(ApiRequest request)
+		{
+			request.ReplyWithJson(JsonConvert.SerializeObject(AvailableFontMetadata));
+		}
+
+		public static IEnumerable<string> SortedListOfFontNames()
+		{
+			var list = new List<string>(Browser.NamesOfFontsThatBrowserCanRender());
+			list.Sort();
+			return list;
+		}
+
+		/// <summary>
+		/// Return a list of FontMetadata objects for all fonts on the system.  This will be SLOW the
+		/// first time it is called, but caches the result so that it will be fast in any later calls.
+		/// </summary>
+		public static IEnumerable<FontMetadata> GetAllFontMetadata()
```

Comment:
Is the return just for test purposes? The normal call should be a background one that just loads up the static results? Would be helpful to say that.